farm-staking/farm-staking/src/claim_only_boosted_staking_rewards.rs

```diff
@@ -49,20 +49,37 @@ pub trait ClaimOnlyBoostedStakingRewardsModule:
         boosted_rewards_payment
     }
 
-    fn migrate_old_farm_positions(&self, caller: &ManagedAddress) {
+    fn migrate_old_farm_positions(&self, caller: &ManagedAddress) -> BigUint {
         let payments = self.call_value().all_esdt_transfers().clone_value();
         let farm_token_mapper = self.farm_token();
         let farm_token_id = farm_token_mapper.get_token_id();
+        let mut migrated_amount = BigUint::zero();
         for farm_position in &payments {
             if farm_position.token_identifier == farm_token_id
                 && self.is_old_farm_position(farm_position.token_nonce)
             {
-                let mut user_total_farm_position = self.get_user_total_farm_position(caller);
-                user_total_farm_position.total_farm_position += farm_position.amount;
-                self.user_total_farm_position(caller)
-                    .set(user_total_farm_position);
+                migrated_amount += farm_position.amount;
             }
         }
+
+        if migrated_amount > 0 {
```

Comment:
why do we have duplicated code ? Can't you add this into base function ?

Reply:
The farm staking contract does not import the base_functions module as there are some variables that are Farm only specific. Maybe we could rewrite this to be more generic.

Reply:
We cannot do that due to Token Attributes. I think it's difficult to rewrite the code write now. The code changes will be huge and we would have to re-audit the whole solution.